### ISSUE TITLE
Problem when using static file management with django + s3

### ISSUE THREAD

**Issue by jamieChae** [summernote/django-summernote#170]
hello.

When doing staticfile management with django + s3
`Django_summernote/templates/django_summernote/widget_iframe_editor.html`
: Problem with static tag in line 27.

`        
 27             $.getScript('{% static "django_summernote/lang/summernote-" %}' + settings.lang + '.min.js');
`

{% static "django_summernote/lang/summernote-" %} 
rendering => "https://{{s3-url}}/static/django_summernote/lang/summernote-"
is not found error.

`
The file 'django_summernote/lang/summernote-' could not be found with <xx.s3ext.StaticS3BotoStorage object at xxxx>.
`

Due to this issue, we have temporarily created a "summernote-" file on s3.

any ideas?
help me.

**Comment by lqez** [comment]
It looks weird. Proper language settings should be popped up via https://github.com/summernote/django-summernote/blob/master/django_summernote/widgets.py#L45-L50

Could you investigate this problem a bit more?

**Comment by larssin** [comment]
I have a similar problem just with `django.contrib.staticfiles.storage.CachedStaticFilesStorage`.
Manually adding a 'django_summernote/lang/summernote-' file to the collected static files solves the problem.
Guess that the `static` template tag is supposed to check for the existence of the file?

I'm running Django 1.10.4 with the settings:
`
settings.DEBUG = False

...

STATIC_URL = '/static/'

STATIC_ROOT = os.path.join(BASE_DIR, "static")

STATICFILES_STORAGE = 'django.contrib.staticfiles.storage.CachedStaticFilesStorage'
...
`

**Comment by umutcoskun** [comment]
I had same problem with larssin. :(

**Comment by riklaunim** [comment]
Just load all supported languange files and let the editor pick. Generating one file name won't end up well.

**Comment by cordery** [comment]
Same here, conflicts with django's [ManifestStaticFilesStorage](https://docs.djangoproject.com/en/1.11/ref/contrib/staticfiles/#manifeststaticfilesstorage), and therefore everything that depends upon it like whitenoise's GzipManifestStaticFilesStorage, etc.  I removed the entire section in order to get the widget to work.

**Comment by mooja** [comment]
Same problem using Django WhiteNoise. Have to use **touch django_summernote/lang/summernote-** during deployment to fix.

**Comment by mrorno** [comment]
I have the same problem with `ManifestStaticFilesStorage`.

See `templates/django_summernote/widget_iframe_editor.html` file, line 27:
```
$.getScript('{% static "django_summernote/lang/summernote-" %}' + settings.lang + '.min.js');
```
`ManifestStaticFilesStorage` will modify `django_summernote/lang/summernote-` to `django_summernote/lang/summernote-.HASH_CODE`. Other custom `StaticFilesStorage` classes will do same kinds of things to generate static url.

And it will cause:
```
ValueError at /summernote/editor/id_body/
Missing staticfiles manifest entry for 'django_summernote/lang/summernote-'
```

My approach is changing `static` use no `settings.lang` javascript value.

First, let's edit `templates/django_summernote/widget_iframe_editor.html` file, line 24-29:
```
// include summernote language pack, synchronously
if( settings.lang != 'en-US' ) {
    $.ajaxSetup({async:false});
    $.getScript('{% static "django_summernote/lang/summernote-" %}' + settings.lang + '.min.js');
    $.ajaxSetup({async:true});
}
```

and replace it with below:
```
{% if lang != 'en-US' %}
    {% with 'django_summernote/lang/summernote-'|add:lang|add:'.min.js' as lang_url %}
        // include summernote language pack, synchronously
        $.ajaxSetup({async: false});
        $.getScript('{% static lang_url %}');
        $.ajaxSetup({async: true});
    {% endwith %}
{% endif %}
```

Now `lang_url` will be `django_summernote/lang/summernote-{{ lang }}.HASH_CODE.min.js`.
`lang` is not provided by view function, so add it.

Open `widgets.py` and copy `_get_proper_language` and paste it to `views.py`. (Or you can import it.) Edit `views.py` to  import `get_language` from `django.utils.translation`

```
from django.utils.translation import get_language, ugettext_lazy as _


def _get_proper_language():
    # Detect language automatically by get_language()
    if not summernote_config['lang']:
        return summernote_config['lang_matches'].get(get_language(), 'en-US')

    return summernote_config['lang']
```

and edit `editor` function:
```
def editor(request, id):
    return render(
        request,
        'django_summernote/widget_iframe_editor.html',
        {
            'id_src': id,
            'id': id.replace('-', '_'),
            'css': (
                summernote_config['default_css'] +
                summernote_config['css']
            ),
            'js': (
                summernote_config['default_js'] +
                summernote_config['js']
            ),
            'disable_upload': summernote_config['disable_upload'],
            'lang': _get_proper_language(),
        }
    )
```

By this edit, we can provide proper language and proper static URL.
May be there is better way to achieve this, hope @lqez will see this 😄

**Comment by lqez** [comment]
I removed hard-coded `{{ STATIC_URL }}` from template files by https://github.com/summernote/django-summernote/pull/220 .
Please try it again with the latest release. Thanks.

**Comment by helb** [comment]
@lqez Seems to be still broken in current version (both 0.8.8.2 and latest master) with Django 1.11.5.
It works with @mrorno 's fix (6e2455c) however.
